gui/mutt_window.c

```diff
@@ -808,3 +809,73 @@ bool window_status_on_top(struct MuttWindow *panel, struct ConfigSubset *sub)
   window_invalidate_all();
   return true;
 }
+
+/**
+ * mutt_window_is_child - Check wether a window is a child of a parent.
+ * @param parent Parent Window
+ * @param win    Window
+ * @retval true  Window is a child of the parent
+ * @retval false Window is not a child of the parent
+ */
+static bool mutt_window_is_child(struct MuttWindow *parent, struct MuttWindow *win)
+{
+  struct MuttWindow *entry = NULL;
+  TAILQ_FOREACH(entry, &parent->children, entries)
+  {
+    if (entry == win)
+      return true;
+  }
+
+  return false;
+}
+
+/**
+ * mutt_window_swap - Swap the position of two windows.
+ * @param parent Parent Window
+ * @param win1   Window
+ * @param win2   Window
+ * @retval true Windows where switched
+ */
+bool mutt_window_swap(struct MuttWindow *parent, struct MuttWindow *win1,
+                      struct MuttWindow *win2)
+{
+  if (!parent || !win1 || !win2)
+    return false;
+
+  struct MuttWindow *win1_next = TAILQ_NEXT(win1, entries);
+  if (win1_next == win2)
+  {
+    // win1 is directly in front of win2, move it behind
+    TAILQ_REMOVE(&parent->children, win1, entries);
```

Comment:
This is a good optimization, but what if win1 and/or win2 aren't children of the parent?

Reply:
If only one window is not a child of the parent we won't reach this. Because then they can't, or at least really shouldn't, be connected.

But you're right, if both are next to each other but not children of the parent this will fail. I'll fix it tomorrow.